src/graphnet/models/task/reconstruction.py

```diff
@@ -39,6 +39,22 @@ def _forward(self, x):
         return angle
 
 
+class DirectionReconstructionWithKappa(Task):
+    """Reconstructs Direction with kappa from The 3DVonMisesFisher Distribution."""
+
+    # Requires 4 features: untransformed points in (x,y,z)-space + kappa estimation.
+    nb_inputs = 4
+
+    def _forward(self, x):
+        # Transform outputs to angle and prepare prediction
+        norm = torch.sqrt(x[:, 0] ** 2 + x[:, 1] ** 2 + x[:, 2] ** 2)
+        kappa = torch.abs(x[:, 3]) + eps_like(x)
+        vec_x = x[:, 0] / norm
+        vec_y = x[:, 1] / norm
+        vec_z = x[:, 2] / norm
+        return torch.stack((vec_x, vec_y, vec_z, kappa), dim=1)
```

Comment:
In the proposed implementation, the normalisation of the `(x,y,z)`-vector is arbitrary, meaning that the task is "overparametrised" (i.e., there is an output, `x[:,3]`, that the model can freely change with no impact on the actual prediction). Would it make sense to take a note from the `AzimuthReconstructionWithKappa` task and instead take the norm of the `(x,y,z)`-vector to represent kappa, like they do in the original paper, I believe? Since kappa is then a norm of a vector, it is automatically non-negative, meaning that we don't need to introduce the discontinuous `torch.abs(...)`.
```suggestion
    # Requires three features: untransformed points in (x,y,z)-space.
    nb_inputs = 3

    def _forward(self, x):
        # Transform outputs to angle and prepare prediction
        kappa = torch.linalg.vector_norm(x, dim=1) + eps_like(x)
        vec_x = x[:, 0] / kappa
        vec_y = x[:, 1] / kappa
        vec_z = x[:, 2] / kappa
        return torch.stack((vec_x, vec_y, vec_z, kappa), dim=1)
```

Reply:
@asogaard Could we perhaps zoom about this?